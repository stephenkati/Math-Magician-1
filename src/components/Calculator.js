import { useState } from 'react';
import calculate from '../logic/calculate';

const Buttons = () => {
  const [total, setTotal] = useState(null);
  const [next, setNext] = useState(null);
  const [operation, setOperation] = useState(null);

  const handleClick = (e) => {
    const answer = calculate({ total, next, operation }, e.target.textContent);
    setTotal(answer.total);
    setNext(answer.next);
    setOperation(answer.operation);
  };

  return (
    <>
      <div className="button output">
        {total}
        {operation}
        {next}
      </div>
      <button onClick={handleClick} type="button" className="button">AC</button>
      <button onClick={handleClick} type="button" className="button">+/-</button>
      <button onClick={handleClick} type="button" className="button">%</button>
      <button onClick={handleClick} type="button" className="button orange">รท</button>
      <button onClick={handleClick} type="button" className="button">7</button>
      <button onClick={handleClick} type="button" className="button">8</button>
      <button onClick={handleClick} type="button" className="button">9</button>
      <button onClick={handleClick} type="button" className="button orange">x</button>
      <button onClick={handleClick} type="button" className="button">4</button>
      <button onClick={handleClick} type="button" className="button">5</button>
      <button onClick={handleClick} type="button" className="button">6</button>
      <button onClick={handleClick} type="button" className="button orange">-</button>
      <button onClick={handleClick} type="button" className="button">1</button>
      <button onClick={handleClick} type="button" className="button">2</button>
      <button onClick={handleClick} type="button" className="button">3</button>
      <button onClick={handleClick} type="button" className="button orange">+</button>
      <button onClick={handleClick} type="button" className="button zero">0</button>
      <button onClick={handleClick} type="button" className="button">.</button>
      <button onClick={handleClick} type="button" className="button orange">=</button>
    </>
  );
};

const Calculator = () => (
  <div className="cal">
    <div className="math">
      <h2 className="math-text">Let&apos;s do some math</h2>
    </div>
    <div className="calculator">
      <Buttons />
    </div>
  </div>
);

export default Calculator;
