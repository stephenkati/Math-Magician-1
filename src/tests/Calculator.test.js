import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('tests for calculator js component', () => {
  test('test to check if calculator matches snapshot', () => {
    const calculatorComponent = renderer
      .create(<Calculator />)
      .toJSON();
    expect(calculatorComponent).toMatchSnapshot();
  });
});
