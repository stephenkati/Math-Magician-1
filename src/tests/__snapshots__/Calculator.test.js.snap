// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for calculator js component test to check if calculator matches snapshot 1`] = `
<div
  className="cal"
>
  <div
    className="math"
  >
    <h2
      className="math-text"
    >
      Let's do some math
    </h2>
  </div>
  <div
    className="calculator"
  >
    <div
      className="button output"
    />
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="button orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="button orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="button orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="button orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="button zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="button orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
